addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/dataexchange/NukiHttpClient.java

```diff
@@ -0,0 +1,170 @@
+        logger.trace("Instantiating NukiHttpClient({})", configuration);
+        this.configuration = configuration;
+        this.httpClient = new HttpClient();
+        long connectTimeout = NukiBindingConstants.CLIENT_CONNECTTIMEOUT;
+        httpClient.setConnectTimeout(connectTimeout);
+    }
+
+    public void startClient() {
+        logger.trace("Starting HttpClient");
+        try {
+            httpClient.start();
+            logger.trace("Started httpClient[{}]", httpClient);
+        } catch (Exception e) {
+            logger.error("Could not start NukiHttpClient! ERROR: {}", e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    public void stopClient() {
+        logger.trace("Stopping HttpClient");
+        try {
+            if (httpClient.isStarted()) {
+                httpClient.stop();
+                logger.trace("Stopped NukiHttpClient[{}]", httpClient);
+            }
+        } catch (Exception e) {
+            logger.error("Could not stop NukiHttpClient! ERROR: {}", e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    public BridgeInfoResponse getBridgeInfo() {
+        logger.debug("NukiHttpClient:getBridgeInfo");
+        String configIp = (String) configuration.get(NukiBindingConstants.CONFIG_IP);
+        BigDecimal configPort = (BigDecimal) configuration.get(NukiBindingConstants.CONFIG_PORT);
+        String configApiToken = (String) configuration.get(NukiBindingConstants.CONFIG_APITOKEN);
+        String uri = String.format(NukiBindingConstants.URI_INFO, configIp, configPort, configApiToken);
+        logger.trace("uri[{}]", uri);
+        try {
+            startClient();
+            ContentResponse contentResponse = httpClient.GET(uri);
+            stopClient();
+            String contentResponseAsString = contentResponse.getContentAsString();
+            logger.trace("contentResponseAsString[{}]", contentResponseAsString);
+            BridgeApiInfoDto bridgeApiInfoDto = new Gson().fromJson(contentResponseAsString, BridgeApiInfoDto.class);
+            BridgeInfoResponse bridgeInfoResponse = new BridgeInfoResponse(contentResponse.getStatus(), "");
+            bridgeInfoResponse.setBridgeInfo(bridgeApiInfoDto);
+            return bridgeInfoResponse;
+        } catch (Exception e) {
+            logger.error("Could not get Bridge Info! ERROR: {}", e.getMessage());
+            if (e instanceof ExecutionException && e.getCause() instanceof HttpResponseException) {
+                return new BridgeInfoResponse(((HttpResponseException) e.getCause()).getResponse().getStatus(),
+                        ((HttpResponseException) e.getCause()).getResponse().getReason());
+            } else {
+                return new BridgeInfoResponse(500, e.getMessage());
+            }
+        }
+    }
+
+    public BridgeLockStateResponse getBridgeLockState(String nukiId) {
+        logger.debug("NukiHttpClient:getBridgeLockState({})", nukiId);
```

Comment:
The method `getBridgeInfo`, `getBridgeLockState`, `getBridgeLockAction` all contain:

* 4 lines to construct the uri
* a startClient, GET, stopClient, converting the resonse a string and logging it
* Then two of them check the statuscode (is it required if you do that when you do a GET, some framework I used respond with a HttpResponseException if it's a 20X)
* Then we do GSON conversion
* The same two of them we handle the things when a non-200 was returned.
* Finally all the methods have more or less the same error handling

With some generics and mapping json result straight to the response class you could easily move all the logic in a single method such that just pass in the uri-format, response class and maybe some logging verb.

Another option could be to split of some method(s) for getting the uri and/or performing the request and returning a JSON. And maybe a method to trigger the processing of the received exception.

Note that if no Checked Exceptions are thrown you might be able to improve the catch line it a little by catching all RuntimeException's (note that the compiler will warn you if you are calling code which throws others exeptions than children of RuntimeException so you can just test it in you IDE)